kentikapi/internal/api_resources/plans_test.go

```diff
@@ -0,0 +1,96 @@
+package api_resources_test
+
+import (
+	"testing"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_connection"
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_resources"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetAllPlans(t *testing.T) {
+	// arrange
+	getResponsePayload := `
+	{
+		"plans":[
+			{
+				"active":true,
+				"bgp_enabled":true,
+				"cdate":"2020-09-03T08:41:57.489Z",
+				"company_id":74333,
+				"description":"Your Free Trial includes 6 devices at a maximum of 1000 fps each. Please contact...",
+				"deviceTypes":[
+					{
+						"device_type":"router"
+					},
+					{
+						"device_type":"host-nprobe-dns-www"
+					}
+				],
+				"devices":[
+					{
+						"id":"77714",
+						"device_name":"testapi_router_minimal_1",
+						"device_type":"router"
+					},
+					{
+						"id":"77720",
+						"device_name":"testapi_dns_minimal_1",
+						"device_type":"host-nprobe-dns-www"
+					},
+					{ 
+						"id":"77724",
+						"device_name":"testapi_router_minimal_postman",
+						"device_type":"router"
+					},
+					{
+						"id":"77715",
+						"device_name":"testapi_router_full_1",
+						"device_type":"router"
+					}
+				],
+				"edate":"2020-09-03T08:41:57.489Z",
+				"fast_retention":30,
+				"full_retention":30,
+				"id":11466,
+				"max_bigdata_fps":30,
+				"max_devices":6,
+				"max_fps":1000,
+				"name":"Free Trial Plan",
+				"metadata":{
+				}
+			}
+		]
+	}`
+	transport := &api_connection.StubTransport{ResponseBody: getResponsePayload}
+	plansAPI := api_resources.NewPlansAPI(transport)
+
+	// act
+	plans, err := plansAPI.GetAll(nil)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	assert.Zero(transport.RequestBody)
+
+	// and response properly parsed
+	require.Equal(1, len(plans))
```

Comment:
I think that it would be more readable to check response with single assert statement:
`assert.Equal(expectedPlans, plans)`

Reply:
TDD results in granular checks as you take small steps adding small tests and implementing only the functionality needed for the test to pass. I can rewrite the tests but imo it's not worthy when short of time.

Reply:
I think that this refactoring would increase test readability, but it won't block my approval